include/sundials/sundials_matrix.h

```diff
@@ -100,6 +104,9 @@ struct _generic_SUNMatrix {
   void *content;
   SUNMatrix_Ops ops;
   SUNContext sunctx;
+#ifdef __cplusplus
+  _generic_SUNMatrix() = default;
+#endif
```

Comment:
Do the other classes need this as well (vector, linear solver, and nonlinear solver)?